app/helpers/concerns/foreman_remote_execution/hosts_helper_extensions.rb

```diff
@@ -22,20 +22,20 @@ def schedule_job_multi_button(*args)
       end
     end
 
-    def rex_host_features(*args)
-      return unless authorized_for(controller: :job_invocations, action: :create)
+    def rex_host_features(host, *_rest)
+      return [] unless can_execute_on_host?(host)
       RemoteExecutionFeature.with_host_action_button.order(:label).map do |feature|
-        link_to(_('%s') % feature.name, job_invocations_path(:host_ids => [args.first.id], :feature => feature.label), :method => :post)
+        link_to(_('%s') % feature.name, job_invocations_path(:host_ids => [host.id], :feature => feature.label), :method => :post)
       end
     end
 
-    def schedule_job_button(*args)
-      return unless authorized_for(controller: :job_invocations, action: :new)
-      link_to(_('Schedule Remote Job'), new_job_invocation_path(:host_ids => [args.first.id]), :id => :run_button, :class => 'btn btn-default')
+    def schedule_job_button(host, *_rest)
+      return unless can_execute_on_host?(host)
+      link_to(_('Schedule Remote Job'), new_job_invocation_path(:host_ids => [host.id]), :id => :run_button, :class => 'btn btn-default')
     end
 
     def web_console_button(host, *args)
-      return unless authorized_for(permission: 'cockpit_hosts', auth_object: host)
+      return unless authorized_for(permission: 'cockpit_hosts', auth_object: host) || !can_execute_on_infrastructure_host?(host)
```

Comment:
when there are two conditions, I find it bit more readable to have it flipped to `if`